### ISSUE TITLE
Cannot add restrictions in complex multiway junction

### ISSUE THREAD

**Issue by srividyacb** [openstreetmap/iD#4968]
[Area in OpenStreetMap](https://www.openstreetmap.org/edit?editor=id#map=20/37.76421/-122.43304)

![image](https://user-images.githubusercontent.com/3423533/38406672-c93c01e2-3993-11e8-8d98-9e1e30e002ea.png)
_Red box in above image showing overlapping restriction arrow in TR editor_

In the same junction cannot add `only` turn-restrcition for highway going straight. The restriction arrow is not going blue even after multiple clicks. 

![tr-issue](https://user-images.githubusercontent.com/3423533/38406734-1be76026-3994-11e8-85ec-4a61c8f2dc3a.gif)

cc @bhousel

**Comment by tyrasd** [comment]
confirming. in the second case, the following exception is thrown:

```
graph.js:48 Uncaught Error: entity r-1 not found
    at coreGraph$$1.entity (graph.js:48)
    at action (delete_relation.js:18)
    at Array.<anonymous> (unrestrict_turn.js:11)
    at _act (history.js:59)
    at _perform (history.js:74)
    at Object.perform (history.js:168)
    at Object.perform (context.js:449)
    at SVGSVGElement.click (restrictions.js:405)
    at SVGSVGElement.<anonymous> (on.js:27)
```

**Comment by tyrasd** [comment]
> adding a no-u-turn restriction at either end of the selected from way causes the no-u-turn icons to be activated at both ends of the way (although only one turn restriction relation is created)

this is actually a regression introduced by 0d262e5bb5572e1ad740a6a97777c54fb0e1da01 for #4928. :thinking:

**Comment by tyrasd** [comment]
no, unfortunately the fix from #4928 doesn't work in this situation (I've originally tested this with a locally checked out master branch, but [http://preview.ideditor.com/master](http://preview.ideditor.com/master/#background=Bing&disable_features=boundaries&id=n581047109&map=20.95/37.76422/-122.43307) shows the same behavior), and introduced the above mentioned "no-u-turn reflection" regression.

**Comment by bhousel** [comment]
> no, unfortunately the fix from #4928 doesn't work in this situation (I've originally tested this with a locally checked out master branch, but http://preview.ideditor.com/master shows the same behavior), and introduced the above mentioned "no-u-turn reflection" regression.

Ah good, thanks for checking 👍
